### PR TITLE
Buffer task outputs instead of printing them interleaved 

### PR DESCRIPTION
**Pull request by SomeHats**
https://user-images.githubusercontent.com/1489520/231740924-0dc73c60-eaa5-499f-8ed8-845844aa663b.mp4

Instead of prefixing task outputs and printing them interleaved which is pretty hard to follow, this diff reworks our logging so that when attached to a TTY, lazyrepo will show an updating summary of the currently executing tasks, then print the logs in a contiguous block when each task finishes. This makes the output much easier to follow. When not attached to a TTY, the output works the same as it does currently.

This works by introducing two new logging modules, one of which is chosen to send all CLI output through. There's the realtime logger (which implements lazyrepo's current output) and a new interactive logger which handles the buffering and updating summary section shown in the video above.

@mrkldshv david mentioned you were planning on looking at logging/log levels - this change is probably going to conflict with that if you've already started working on it, so sorry about that! We should be able to figure out a solution to log-levels that integrates with these logging modules.

### CODE REVIEW

File: src/logger/InteractiveLogger.js [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+  }
+  if (a.startedAt > b.startedAt) {
+    return 1
+  }
+  if (a.prefix < b.prefix) {
+    return -1
+  }
+  if (a.prefix > b.prefix) {
+    return 1
+  }
```
Comment (ds300) [inline src/logger/InteractiveLogger.js:261]: Idea for the future that may totally suck but I'd like to try anyway:

```
<list of completed tasks, ordered by completion time + prefix (earliest first)>
<blank line>
<list of running tasks, ordered by start time + prefix>
```

Then when tasks complete they kinda pop up into the upper section. Could be a fun juicy flourish. Also might just be confusing/disorienting.

File: src/logger/InteractiveLogger.js [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+      this.moveCursorToAboveTasksSection()
+      this.tty.clearScreenDown()
+      task.status = status
+      const lastLine = lastNonEmptyLineIfPossible(message)
+      if (lastLine) {
+        task.lastLogLine = lastLine
+      }
+
+      bufferedLogLines.push(message)
+      this.tty.write(color.fg(':: ') + color.bg().bold(` ${taskName} `) + color.fg(' ::') + '\n')
```
Comment (ds300) [inline src/logger/InteractiveLogger.js:207]: This feels kinda visually intense. Might try tweaking it after this gets merged, but if you wanna give it a go first please do :)

**Review by ds300 — APPROVED** [review]
I LOVE THIS. It's so much easier to follow things now. Gonna push a lint fix, feel free to merge when ready.

**Review by ds300 — APPROVED** [review]

**Comment by ds300** [comment]
noticed an edge case where task failure doesn't result in the completion being invoked to output the buffered lines, pushing a fix

File: src/logger/InteractiveLogger.js (hunk 1 above)
Comment (SomeHats) [reply src/logger/InteractiveLogger.js:261]: I tried this before but found things jumping around like that to be a little distracting / hard to follow. Keeping things stable seemed the easiest to follow

**Review by ds300 — APPROVED** [review]
